[readline(win32)] Readline package not found by CMake

**Describe the bug**
Well, when installing the readline package (readline-win32), vcpkg doesn't give the hint on how to import this library into the project.

![image](https://user-images.githubusercontent.com/42288418/177405375-60e66cd9-3842-45d2-8af9-47a39ddfb5e6.png)

When trying to guess the name of the library, CMake warns by Intelisense that it did not find the library.
![image](https://user-images.githubusercontent.com/42288418/177405556-438fa3a9-913a-4a65-bc5d-1addec05b184.png)

When compiling, another error appears.

1> [CMake] CMake Error at D:/Programacao/Repositorios/vcpkg/scripts/buildsystems/vcpkg.cmake:824 (_find_package):
1> [CMake]   Could not find a package configuration file provided by "readline" with any
1> [CMake]   of the following names:
1> [CMake] 
1> [CMake]     readlineConfig.cmake
1> [CMake]     readline-config.cmake
1> [CMake] 
1> [CMake]   Add the installation prefix of "readline" to CMAKE_PREFIX_PATH or set
1> [CMake]   "readline_DIR" to a directory containing one of the above files.  If
1> [CMake]   "readline" provides a separate development package or SDK, be sure it has
1> [CMake]   been installed.
1> [CMake] Call Stack (most recent call first):
1> [CMake]   CMakeLists.txt:14 (find_package)
1> [CMake] -- Configuring incomplete, errors occurred!

**Environment**
- OS: Windows 10
- Compiler: MSVC

**To Reproduce**
Steps to reproduce the behavior:
1. ./vcpkg install readline-win32
2. Try to import this library into your project (somehow why won't vpckg tell you what the name of the library is for you to import via find_package())
3. Compile case you can import

**Expected behavior**
I can import the library, use it normally and compile it with my project.

**Failure logs**
It doesn't have an error log other than the Visual Studio console output.

**Additional context**
Nothing.

--- a/ports/readline-win32/CMakeLists.txt
+++ b/ports/readline-win32/CMakeLists.txt
@@ -56,11 +56,11 @@ install(FILES ${headers} DESTINATION include/readline)
 target_include_directories(readline PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}> $<INSTALL_INTERFACE:include>)
 
 install(TARGETS readline
-    EXPORT unofficial-readline-win32Config
+    EXPORT unofficial-readline-win32-config
     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
-install(EXPORT unofficial-readline-win32Config
-    NAMESPACE unofficial::readline::
+install(EXPORT unofficial-readline-win32-config
+    NAMESPACE unofficial::readline-win32::
     DESTINATION share/unofficial-readline-win32)

--- a/ports/readline-win32/CMakeLists.txt
+++ b/ports/readline-win32/CMakeLists.txt
@@ -50,7 +50,17 @@ add_library(readline
 	mbutil.c
     support/wcwidth.c)
 
+file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
+install(FILES ${headers} DESTINATION include/readline)
+
+target_include_directories(readline PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}> $<INSTALL_INTERFACE:include>)
+
 install(TARGETS readline
-    RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib)
+    EXPORT unofficial-readline-win32Config
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+install(EXPORT unofficial-readline-win32Config
+    NAMESPACE unofficial::readline::
+    DESTINATION share/unofficial-readline-win32)

--- a/ports/readline-win32/portfile.cmake
+++ b/ports/readline-win32/portfile.cmake
@@ -6,21 +6,19 @@ vcpkg_from_github(
     HEAD_REF master
 )
 
-file(COPY ${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt DESTINATION ${SOURCE_PATH}/src/readline/5.0/readline-5.0-src)
-file(COPY ${CMAKE_CURRENT_LIST_DIR}/config.h DESTINATION ${SOURCE_PATH}/src/readline/5.0/readline-5.0-src)
+file(COPY "${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt" DESTINATION "${SOURCE_PATH}/src/readline/5.0/readline-5.0-src")
+file(COPY "${CMAKE_CURRENT_LIST_DIR}/config.h" DESTINATION "${SOURCE_PATH}/src/readline/5.0/readline-5.0-src")
 
-vcpkg_configure_cmake(
-    SOURCE_PATH ${SOURCE_PATH}/src/readline/5.0/readline-5.0-src
-    PREFER_NINJA
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}/src/readline/5.0/readline-5.0-src"
 )
 
-vcpkg_install_cmake()
+vcpkg_cmake_install()
 
-# Copy headers
-file(MAKE_DIRECTORY ${CURRENT_PACKAGES_DIR}/include/readline)
-file(GLOB headers "${SOURCE_PATH}/src/readline/5.0/readline-5.0-src/*.h")
-file(COPY ${headers} DESTINATION ${CURRENT_PACKAGES_DIR}/include/readline)
+vcpkg_cmake_config_fixup(PACKAGE_NAME unofficial-readline-win32)
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share" "${CURRENT_PACKAGES_DIR}/debug/include")
 
 vcpkg_copy_pdbs()
 
-file(INSTALL ${SOURCE_PATH}/src/readline/5.0/readline-5.0-src/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+file(INSTALL "${SOURCE_PATH}/src/readline/5.0/readline-5.0-src/COPYING" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)

--- a/ports/readline-win32/vcpkg.json
+++ b/ports/readline-win32/vcpkg.json
@@ -1,8 +1,19 @@
 {
   "name": "readline-win32",
   "version": "5.0",
-  "port-version": 4,
+  "port-version": 5,
   "description": "Implementation of readline for Windows Desktop",
   "homepage": "https://github.com/lltcggie",
-  "supports": "windows & !uwp"
+  "license": null,
+  "supports": "windows & !uwp",
+  "dependencies": [
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
+      "host": true
+    }
+  ]
 }

--- a/ports/readline-win32/vcpkg.json
+++ b/ports/readline-win32/vcpkg.json
@@ -3,7 +3,7 @@
   "version": "5.0",
   "port-version": 5,
   "description": "Implementation of readline for Windows Desktop",
-  "homepage": "https://github.com/lltcggie",
+  "homepage": "https://github.com/lltcggie/readline",
   "license": null,
   "supports": "windows & !uwp",
   "dependencies": [

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -6198,7 +6198,7 @@
     },
     "readline-win32": {
       "baseline": "5.0",
-      "port-version": 4
+      "port-version": 5
     },
     "readosm": {
       "baseline": "1.1.0a",

--- a/versions/r-/readline-win32.json
+++ b/versions/r-/readline-win32.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "2507f88f520d5472ec37644c97aa860d471eaff6",
+      "version": "5.0",
+      "port-version": 5
+    },
+    {
       "git-tree": "b1ed33b8ec37a1cf44926c9954b6943bce302b70",
       "version": "5.0",
       "port-version": 4

--- a/versions/r-/readline-win32.json
+++ b/versions/r-/readline-win32.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "2507f88f520d5472ec37644c97aa860d471eaff6",
+      "git-tree": "19f9684f55a9aa01b5612a5ebec47960f51bacdd",
       "version": "5.0",
       "port-version": 5
     },

